sdk/identity/identity/src/credentials/chainedTokenCredential.ts

```diff
@@ -32,14 +33,22 @@ export class ChainedTokenCredential implements TokenCredential {
     let token = null;
     const errors = [];
 
+    const { span, options: newOptions } = createSpan("ChainedTokenCredential-getToken", options);
+
     for (let i = 0; i < this._sources.length && token === null; i++) {
       try {
-        token = await this._sources[i].getToken(scopes, options);
+        token = await this._sources[i].getToken(scopes, newOptions);
       } catch (err) {
+        span.setStatus({
```

Comment:
The purpose of this loop is to try each credential in the chain until one is able to authenticate successfully.  If there are 3 credentials in the chain and the first two throw an exception but the last succeeds, shouldn't the span be considered successful?  Maybe we should only trace in line 44/53 below where the `AggregateAuthenticationError` gets thrown and use the message from that exception instead.

Reply:
The span is specific to the work it wraps. I think we do actually want to see that some spans failed, as it's the trace that will ultimately succeed (if there's a top-level span that wraps the chain, that could indicate success or failure as well).

For example, it can be useful for the customer to see that the 1st credential always fails, or only the 1st and 3rd pass. Then they could create their own chain if they wanted. Or maybe they think they are using one credential but it turns out that one is failing and they always fall back to another.

Reply:
Oh I see, the span is started outside the loop, not inside. (Missed that context when just reading the comment).